tools/update_quest_data.py

```
@@ -0,0 +1,439 @@
+        if not quests_source:
+            for spot in war_data.get("spots", []):
+                quests_source.extend(spot.get("quests", []))
+
+        for quest in quests_source:
+            # 筛选 Free 副本: afterClear == "repeatLast" 且有多个 phase
+            after_clear = quest.get("afterClear", "")
+            if after_clear == "repeatLast" and quest.get("phases"):
+                free_quests.append(quest)
+
+        if not free_quests:
+            print(f"无 Free 副本")
```

Ruff (0.15.10)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)